tensorflow/core/kernels/segment_reduction_ops_impl.h

```diff
@@ -957,6 +957,125 @@ class SparseSegmentSqrtNGradOp
                                                      true /*is_sqrtn*/) {}
 };
 
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
+template <class Device, class T, class Tindex, bool has_num_segments>
+class SparseSegmentSumGpuOp : public AsyncOpKernel {
+ public:
+  explicit SparseSegmentSumGpuOp(OpKernelConstruction* context)
+      : AsyncOpKernel(context){};
+
+  void ComputeAsync(OpKernelContext* context, DoneCallback done) override {
+    const Tensor& input_data = context->input(0);
+    const Tensor& indices = context->input(1);
+    const Tensor& segment_ids = context->input(2);
+
+    OP_REQUIRES_ASYNC(context, TensorShapeUtils::IsVector(indices.shape()),
+                      errors::InvalidArgument("indices should be a vector."),
+                      done);
+    const int64 num_indices = indices.NumElements();
+
+    OP_REQUIRES_ASYNC(
+        context, TensorShapeUtils::IsVector(segment_ids.shape()),
+        errors::InvalidArgument("segment_ids should be a vector."), done);
+
+    OP_REQUIRES_ASYNC(
+        context, num_indices == segment_ids.NumElements(),
+        errors::InvalidArgument("indices and segment_ids should have"
+                                "same length."),
+        done);
+
+    ScratchSpace<Index> output_rows_host(context, 1, /* on_host */ true);
+    auto stream = context->op_device_context()->stream();
+
+    if (has_num_segments) {
+      const Tensor& num_segments = context->input(3);
+      se::DeviceMemoryBase num_segments_device(
+          const_cast<Tensor&>(num_segments).template flat<Index>().data()
+      );
+      OP_REQUIRES_ASYNC(
+          context,
+          stream
+              ->ThenMemcpy(output_rows_host.mutable_data(), num_segments_device, sizeof(Index)).ok(),
+          errors::Internal(
+              "SparseSegmentSumGpuOp: failed to copy num_segments to host."),
+          done);
+    } else {
+      se::DeviceMemoryBase last_segment_id_on_device(
+          const_cast<Tensor&>(segment_ids).template flat<Index>().data() +
+          (num_indices - 1));
+      OP_REQUIRES_ASYNC(
```

Comment:
Please use `OP_REQUIRES_OK_ASYNC`.

Reply:
I checked the `OP_REQUIRES_OK_ASYNC` macro. It seems that a `Status` condition is needed. But `ThenMemcpy` returns a `Stream`, which only have an `ok()` method with `bool` return value. It's a little confusing with `Status::ok()`.